### PR TITLE
Reduced code verbosity with `final var`/`var`

### PR DESCRIPTION
**Pull request by MarcinOrlowski**
This is the bigger refactoring, but it shall be beneficial in longer run. Changes introduced by this PR:

* Replaced explicit variable type declaration with `final var` (or `var`) whenever possible (in affected files - this is not touching whole code base)
* Marked interface method implementations with @Override
* Reduced number of imports

I changed only code that I was confident `var `results is exactly the same primitive type, direct subclass etc. So all places we had i.e.

    long foo = .....

that was suggested to become taken as `int foo =...` is left unaltered. 

I also left all the 

    Object foo = ...

things. This is bad to have so generic declaration, but this definitely have to be fixed and narrowed down to some interface or so, but that's not part of this PR.

I occasionally also  renamed local variables here and there and some method **arguments** too (leaving method names mostly unaltered, with a few exceptions for some local/private methods).

### CODE REVIEW

**Comment by davidhutchens** [comment]
The title of the PR suggests it is reducing verbose code. That is true when replacing "uglyLongTypeInfo x = new uglyLongTypeInfo()" with "var x = new uglyLongTypeInfo()". But replacing "for (simpleType x : someCollection)" with "for (var x : someCollection)" doesn't reduce verbosity so much as it obscures the type from the reader, since the reader will now have to look at the definition of someCollection to determine the type of x, assuming the definition of someCollection is not similarly obscured.

I'd suggest looking at each instance and asking which way is most easily understood by someone looking at the code.

**Comment by MarcinOrlowski** [comment]
> But replacing "for (simpleType x : someCollection)" with "for (var x : someCollection)" doesn't reduce verbosity so much as it obscures the type from the reader, since the reader will now have to look at the definition of 

Thanks for the feedback. I do agree that  that type of `x` is now hidden but the question is if that's really the problem. Do you care the type or if variable name makes sense it's sufficient for you? Because we have tons of one-letter vars (esp. in loops like the one you shown), so maybe it would be perfectly sufficient to rename cryptic `x` to something more self-explanatory. Also depending all modern IDEs will show you inferred type either in-lined or on-hoover. 

Finally, you can also use plugins like [Advanced Java Folding](https://medium.com/@andrey_cheptsov/making-java-code-easier-to-read-without-changing-it-adeebd5c36de) that can significantly change how the code is shown in editor. The main goal was to reduce repetitive declarations in the first place. As side effect, I also `final`ized whatever I could.

To sum up - I will try to go thru the code altered in by this PR and rename one letter vars as I believe that would be probably sufficient solution.

**Comment by MarcinOrlowski** [comment]
> But replacing "for (simpleType x : someCollection)" with "for (var x : someCollection)" doesn't reduce verbosity so much as it obscures the type from the reader, since the reader will now have to look at the definition of 

Thanks for the feedback. I do agree that  that type of `x` is now hidden but the question is if that's really the problem. Do you care the type or if variable name makes sense it's sufficient for you? Because we have tons of one-letter vars (esp. in loops like the one you shown), so maybe it would be perfectly sufficient to rename cryptic `x` to something more self-explanatory. Also all modern IDEs will show you inferred type either in-lined or on-hoover. 

Finally, you can also use plugins like [Advanced Java Folding](https://medium.com/@andrey_cheptsov/making-java-code-easier-to-read-without-changing-it-adeebd5c36de) that can significantly change how the code is shown in editor. The main goal was to reduce repetitive declarations in the first place. As side effect, I also `final`ized whatever I could.

To sum up - I will try to go thru the code altered in by this PR and rename one letter vars first and then we see.  How you find that?

**Comment by BFH-ktt1** [comment]
@MarcinOrlowski , I see that there are conflicts, could you look into them?

**Comment by MarcinOrlowski** [comment]
@BFH-ktt1 In sync now.

**Comment by BFH-ktt1** [comment]
@MarcinOrlowski , thanks! will merge as soon as the checks have finished
